### PR TITLE
Reduce number of mandatory fields in NXcxi_ptycho

### PR DESCRIPTION
**Pull request by aaron-parsons**
Hi @vasole,@prjemian and @markbasham,

In response to #647, @vincefn and I have been talking over the past couple of days.

We have agreed that making these fields optional rather than mandatory is a good idea and will probably improve uptake of the definition. Since the structure of the definition doesn't change, this is a backwards compatible change.

Let me know if theres anything else that is needed to get this into the main repo.

Thanks,

Aaron

### CODE REVIEW

**Comment by vasole** [comment]
If a user community is providing an HDF5 based format, I see little interest in translating that format to NeXus. All what NeXus should do is to provide a container for it (typically an NXsubentry) in order

- to be able to refer to the original community supplied documentation and not a translation
- to allow software developers to easily support the format/definition irrespective of following the original or the NeXusfied version.

**Comment by vasole** [comment]
Well, previous comment probably should be at the PR level.

File: contributed_definitions/NXcxi_ptycho.nxdl.xml [hunk 1]
```diff
@@ -62,12 +62,10 @@
           Application definition for a ptychography experiment. This is compatible with CXI from version 1.6 (discussed with Filippe Maia) if this application definition
```
Comment (prjemian) [inline contributed_definitions/NXcxi_ptycho.nxdl.xml]: Is this text necessary for the documentation?

> (discussed with Filippe Maia)

File: contributed_definitions/NXcxi_ptycho.nxdl.xml [hunk 2]
```diff
@@ -62,12 +62,10 @@
...
           is put at the top "entry" level. Above this a "cxi_version" field should be defined. The CXI format is name based, rather than class based, and so it is important
           to pay attention to the naming convention to be CXI compatible. There are duplications due to the format merger. These should be achieved by SoftLinks,
```
Comment (prjemian) [inline contributed_definitions/NXcxi_ptycho.nxdl.xml]: NeXus uses [HDF5 Hard Links and the `@target` attribute](http://download.nexusformat.org/doc/html/design.html#index-14) to choose which HDF5 object reference is to be considered the original.  If this *NXcxi_ptycho* definition will set a precedent for using HDF5 Soft Links, ratification of this application definition by the NIAC may involve some discussion.

**Comment by prjemian** [comment]
@aaron-parsons: +1 for reducing the number of required components.  In a NeXus application definition, everything is required (`minOccurs=1`) unless explicitly declared otherwise.

**Comment by aaron-parsons** [comment]
Hi Pete,

Thanks for the comments.

I'm on a beamtime at the moment so have very little time to look at this, but will be coming back to it next week.

Aaron

**Comment by prjemian** [comment]
@aaron-parsons : Comments?

File: contributed_definitions/NXcxi_ptycho.nxdl.xml (hunk 1 above)
Comment (aaron-parsons) [reply contributed_definitions/NXcxi_ptycho.nxdl.xml]: I've now removed this text.

File: contributed_definitions/NXcxi_ptycho.nxdl.xml (hunk 2 above)
Comment (aaron-parsons) [reply contributed_definitions/NXcxi_ptycho.nxdl.xml]: I changed this text. Any form of linking is fine, so the user should use the standard for nexus.

File: contributed_definitions/NXcxi_ptycho.nxdl.xml [hunk 3]
```diff
@@ -59,15 +59,13 @@
     </symbols>
 
     <doc>
-          Application definition for a ptychography experiment. This is compatible with CXI from version 1.6 (discussed with Filippe Maia) if this application definition
+          Application definition for a ptychography experiment. This is compatible with CXI from version 1.6 if this application definition
```
Comment (prjemian) [inline contributed_definitions/NXcxi_ptycho.nxdl.xml]: This should be a complete sentence on only one line (used as a summary of this NXDL), to be consistent with the other NXDL files.  As it stands now, this image shows how the current summary will appear:

Clipboard01

Compare with current NeXus documentation: http://download.nexusformat.org/doc/html/classes/contributed_definitions/index.html#contributed-definitions

Comment (prjemian) [reply contributed_definitions/NXcxi_ptycho.nxdl.xml]: *complete sentence* - not so much it seems

possibly:
*definition for a ptychography experiment, compatible with CXI from version 1.6*

File: contributed_definitions/NXcxi_ptycho.nxdl.xml [hunk 4]
```diff
@@ -59,15 +59,13 @@
...
           is put at the top "entry" level. Above this a "cxi_version" field should be defined. The CXI format is name based, rather than class based, and so it is important
-          to pay attention to the naming convention to be CXI compatible. There are duplications due to the format merger. These should be achieved by SoftLinks, 
-          with hdf5 Virtual Dataset being used to restructure any data that needs to be remapped. 
+          to pay attention to the naming convention to be CXI compatible. There are duplications due to the format merger. These should be achieved by linking, 
+          with hdf5 Virtual Dataset being used to restructure any data that needs to be remapped. To be fully CXI compatible, all units (including energy) must be in SI units. 
           
           An example here is that CXI expects the data to always to have shape (npts_x*npts_y, frame_size_x, frame_size_y). For nexus this is only true for arbitrary scan paths
```
Comment (prjemian) [inline contributed_definitions/NXcxi_ptycho.nxdl.xml]: remove the extra " to "

**Pull request merged**